### ISSUE TITLE
Archive log files to S3 for easy access to failure logs

### ISSUE THREAD

**Issue by fossell** [NCAR/wrfcloud#173]
## Describe the New Feature ##
There is currently no way to really know why a run fails if it fails within the runtime module. To debug, one must launch a cluster manually and rerun that failed job id and search through the logs to identify the root cause.  A short term fix is to automatically archive the log files to the S3 bucket so they are easy to access should the need arise. A longer term solution should be identified as time allows in a separate issue. 

Tasks include:

- [ ] Code to bundle log files and archive on S3
- [ ] Adding documentation in UG describing these files

### Acceptance Testing ###
*List input data types and sources.*
*Describe tests required for new functionality.*

### Time Estimate ###
*Estimate the amount of work required here.*
*Issues should represent approximately 1 to 3 days of work.*

### Sub-Issues ###
Consider breaking the new feature down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
*List relevant project deadlines here or state NONE.*

### Assignee ###
- [ ] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [ ] Select **component(s)**
- [ ] Select **priority**

### Projects and Milestone ###
- [ ] Select **Project**
- [ ] Select **Milestone** as the next official version or **Backlog of Development Ideas**

## New Feature Checklist ##
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>/<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)**, **Project**, and **Development** issue
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

**Comment by fossell** [comment]
@hahnd - Here are the important log files to archive for each component:

**geogrid:** geogrid.log
**ungrib:** ungrib.log
**metgrid:** metgrid.log
**real:** Currently always run in parallel, so look for real*.log and rsl* files. There will be one rsl.error.xxxx and rsl.out.xxxx file for each processor used. Typically the most important info is in rsl.out.0000, but sometimes in others. Best to tar them all up.
**wrf:** Currently always run in parallel, so look for wrf*.log and rsl* files. There will be one rsl.error.xxxx and rsl.out.xxxx file for each processor used. Typically the most important info is in rsl.out.0000, but sometimes in others. Best to tar them all up. Note that both real and wrf output to the same named rsl files. When tarring up, keep real and wrf rsl files separate and name them something meaningful, e.g. rsl_real.tar or something.
**upp:** unipost*.log, unipost_*.out
